src/it/java/org/jboss/tools/intellij/kubernetes/utils/GlobalUtils.java

```diff
@@ -0,0 +1,349 @@
+        } catch (AWTException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static BufferedImage getScreenshotAsBufferedImage() throws AWTException {
+        Rectangle fullscreenRect = new Rectangle(Toolkit.getDefaultToolkit().getScreenSize());
+        BufferedImage screenshot = new Robot().createScreenCapture(fullscreenRect);
+        return screenshot;
+    }
+
+    private static String getTimeNowAsString(String format) {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern(format);
+        LocalDateTime now = LocalDateTime.now();
+        return dtf.format(now);
+    }
+
+    private static boolean didTheProjectImportFinish(RemoteRobot remoteRobot) {
+        try {
+            remoteRobot.find(ComponentFixture.class, byXpath("//div[@class='EngravedLabel']"));
+        } catch (WaitForConditionTimeoutException e) {
+            return true;
+        }
+        return false;
+    }
+
+    private static boolean didAllTheBgTasksFinish(RemoteRobot remoteRobot) {
+        for (int i = 0; i < 5; i++) {
+            final IdeStatusBarFixture ideStatusBarFixture = remoteRobot.find(IdeStatusBarFixture.class);
+            List<RemoteText> inlineProgressPanelContent = ideStatusBarFixture.inlineProgressPanel().findAllText();
+            if (!inlineProgressPanelContent.isEmpty()) {
+                return false;
+            }
+
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+        return true;
+    }
+
+    private static int getNumberOfProjectLinks(RemoteRobot remoteRobot) {
+        final WelcomeFrameDialogFixture welcomeFrameDialogFixture = remoteRobot.find(WelcomeFrameDialogFixture.class, Duration.ofSeconds(10));
+        try {
+            ComponentFixture cf = welcomeFrameDialogFixture.find(ComponentFixture.class, byXpath("//div[@accessiblename='Recent Projects' and @class='MyList']"));
+            int numberOfProjectsLinks = cf.findAllText().size() / 2;    // 2 items per 1 project link (project path and project name)
+            return numberOfProjectsLinks;
+        } catch (WaitForConditionTimeoutException e) {
+            // the list with accessible name 'Recent Projects' is not available -> 0 links in the 'Welcome to IntelliJ IDEA' dialog
+            return 0;
+        }
+    }
+
+    public static void waitUntilIntelliJStarts(int port) {
+        waitFor(Duration.ofSeconds(600), Duration.ofSeconds(3), "The IntelliJ Idea did not start in 10 minutes.", () -> isIntelliJUIVisible(port));
+    }
+
+    private static boolean isIntelliJUIVisible(int port) {
+        return isHostOnIpAndPortAccessible("127.0.0.1", port);
+    }
+
+    private static boolean isHostOnIpAndPortAccessible(String ip, int port) {
+        SocketAddress sockaddr = new InetSocketAddress(ip, port);
```

Comment:
`sockaddr` seems not used? Maybe it should be passed to the Socket?

Reply:
it used in "try catch" or what do you mean?

Reply:
oh, my bad, only looked at the constructor, missed that in the eventual try/catch-block, the address is being used. Resolving.